include/ddsrouter/communication/Track.hpp

```diff
@@ -193,20 +189,25 @@ class Track
     std::atomic<bool> is_data_available_;
```

Comment:
The following problem might occur: in `transmit_`, just after taking the last sample from the reader a new sample arrives, but it is not read until new data arrives because `no_more_data_available_` is called after `data_available`.

I believe this might be solved by making `is_data_available_` to have three possible values, like for example: no_data, data_available and transmitting_available_data. At the beginning of `transmit_`, change the state of the variable, and also in `data_available` callback depending on its current value.